src/view/output-view.js

```diff
@@ -7,62 +7,68 @@ const ACCESS_CONTROL_EXCEPTION = "java.security.AccessControlException";
+const BUG_REPORT_MESSAGE = 'Hey! It seems you just found a bug! \uD83D\uDC1E\n' +
+  `Please go here -> http://kotl.in/issue to submit it ` +
   `to the issue tracker and one day we fix it, hopefully \uD83D\uDE09\n` +
-  `✅ Don't forget to attach code to the issue${ANGLE_BRACKETS_LEFT_HTML}/errStream${ANGLE_BRACKETS_RIGHT_HTML}\n`;
+  `✅ Don't forget to attach code to the issue\n`;
 
-export function processJVMOutput(output, theme) {
-  let processedOutput = processingHtmlBrackets(output); // don't need to escape `&`
-  return processedOutput
-    .split(BUG_FLAG).join(BUG_REPORT_MESSAGE)
-    .split(`${ANGLE_BRACKETS_LEFT_HTML}outStream${ANGLE_BRACKETS_RIGHT_HTML}`).join(`<span class="standard-output ${theme}">`)
-    .split(`${ANGLE_BRACKETS_LEFT_HTML}/outStream${ANGLE_BRACKETS_RIGHT_HTML}`).join("</span>")
-    .split(`${ANGLE_BRACKETS_LEFT_HTML}errStream${ANGLE_BRACKETS_RIGHT_HTML}`).join(`<span class="error-output ${theme}">`)
-    .split(`${ANGLE_BRACKETS_LEFT_HTML}/errStream${ANGLE_BRACKETS_RIGHT_HTML}`).join("</span>");
+export function processJVMStdout(output, theme) {
+  const processedOutput = escapeBrackets(output);
+  return `<span class="standard-output ${theme}">${processedOutput}</span>`
 }
 
-export function processJUnitResults(data, onTestPassed, onTestFailed) {
-  let result = "";
-  let totalTime = 0;
-  let passed = true;
-  if (isEmptyObject(data)) return NO_TEST_FOUND;
-  for (let testClass in data) {
-    let listOfResults = arrayFrom(data[testClass]);
-    result += listOfResults.reduce((previousTest, currentTest) => {
-      totalTime = totalTime + (currentTest.executionTime / 1000);
-      if (currentTest.status === TEST_STATUS.ERROR.value || currentTest.status === TEST_STATUS.FAIL.value) passed = false;
-      switch (currentTest.status) {
-        case TEST_STATUS.FAIL.value:
-          return previousTest + buildOutputTestLine(TEST_STATUS.FAIL.text, currentTest.methodName, currentTest.comparisonFailure.message);
-        case TEST_STATUS.ERROR.value:
-          return previousTest + buildOutputTestLine(TEST_STATUS.ERROR.text, currentTest.methodName, currentTest.exception.message);
-        case TEST_STATUS.PASSED.value:
-          return previousTest + buildOutputTestLine(TEST_STATUS.PASSED.text, currentTest.methodName, "");
-      }
-    }, "");
+export function processJVMStderr(output, theme) {
+  if (output === BUG_FLAG) {
+    output = BUG_REPORT_MESSAGE
+  }
+  const processedOutput = escapeBrackets(output);
+  return `<span class="error-output ${theme}">${processedOutput}</span>`
+}
+
+export function processJUnitTotalResults(testResults, onTestPassed, onTestFailed) {
+  if (testResults.testsRun === 0) {
+    return NO_TEST_FOUND
+  }
+  if (testResults.success) {
+    if (onTestPassed) onTestPassed()
+  } else {
+    if (onTestFailed) onTestFailed()
+  }
+  return `<div class="test-time">Total test time: ${testResults.totalTime}s</div>`
+}
+
+export function processJUnitTestResult(testRunInfo, testResults) {
```

Comment:
@lyubortk actually i don't understand what is happend with test output. Can you explain me differences in this output. It's too much changes not clear for me :(

Reply:
I had to rewrite that because each test result must be handled separately.